packages/ra-ui-materialui/src/button/Button.spec.tsx

```diff
@@ -2,7 +2,8 @@ import { render, cleanup } from '@testing-library/react';
 import * as React from 'react';
 import expect from 'expect';
 import { TestContext } from 'ra-core';
-import { createMuiTheme, ThemeProvider } from '@material-ui/core';
+import { ThemeProvider } from '@material-ui/core';
```

Comment:
Shouldn't this be imported from `@material-ui/core/styles` as well? Same for all other `ThemeProvider` imports

Reply:
That's an interesting case: `ThemeProvider` normally lives in `@material-ui/styles`, which means it's re-exported from `@material-ui/core` and is in theory subject to the same problem we are talking about here. However, I've failed to confirm this so far, still looking into the source code. It doesn't seem to hurt compilation anyway, at least not for a minimal project, but I'll keep digging.

Reply:
OK, found it. [`@material-ui/core/styles`](https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/styles/index.d.ts#L26) exports a couple of exports from `@material-ui/styles`, while [`@material-ui/core`](https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/index.d.ts#L57) re-exports everything from `@material-ui/core/styles`. Now to find out why this works and e.g. `makeStyles` doesn't.

Reply:
I don't think I get it fully, but it seems that `ThemeProvider` is resolved correctly with the `babel` rules because at the end of the import chain it's not a part of `@material-ui/core` but of `@material-ui/styles`, which is not getting transformed by `babel`. In theory, everything under `core/styles` should probably be part of the rule:

```
colorManipulator.js
createBreakpoints.js
createMixins.js
createMuiStrictModeTheme.js
createMuiTheme.js
createPalette.js
createSpacing.js
createStyles.js
createTypography.js
cssUtils.js
defaultTheme.js
makeStyles.js
MuiThemeProvider.js
responsiveFontSizes.js
shadows.js
shape.js
styled.js
transitions.js
useTheme.js
withStyles.js
withTheme.js
zIndex.js
```

But the question is probably how to keep up with changes. I think `createMuiTheme` and `makeStyles` could be enough for starters and we'll have to see what happens with MUI v5, this problem might disappear in the process.